### PR TITLE
add docs site

### PR DESCRIPTION
**Pull request by chdsbd**
The current README is hard to navigate and discourages adding more examples. Having a documentation website should allow us to provide more detailed docs.

### CODE REVIEW

File: docs/static/img/undraw_docusaurus_tree.svg [hunk 1]
```diff
@@ -0,0 +1 @@
...84463,3.60248,3.60248,0,0,1,.06636-.65027,3.82638,3.82638,0,0,1,.18447-.62373,3.48858,3.48858,0,0,1,.30656-.57064,3.197,3.197,0,0,1,.91436-.91568,3.44685,3.44685,0,0,1,.572-.30523,3.344,3.344,0,0,1,.62372-.18578,3.06907,3.06907,0,0,1,1.30053,0,3.22332,3.22332,0,0,1,1.19436.491,5.02835,5.02835,0,0,1,.50429.41139,4.8801,4.8801,0,0,1,.41139.50429,3.38246,3.38246,0,0,1,.30522.57064,3.47806,3.47806,0,0,1,.25215,1.274A3.36394,3.36394,0,0,1,426.36,446.865a5.02708,5.02708,0,0,1-.50429.4114,3.3057,3.3057,0,0,1-1.84463.55737m26.54143-1.65884a3.38754,3.38754,0,0,1-2.35024-.96877,5.04185,5.04185,0,0,1-.41007-.50428,3.27532,3.27532,0,0,1-.55737-1.84463,3.38659,3.38659,0,0,1,.96744-2.34892,5.02559,5.02559,0,0,1,.50429-.41139,3.44685,3.44685,0,0,1,.572-.30523,3.3432,3.3432,0,0,1,.62373-.18579,3.06952,3.06952,0,0,1,1.30052,0,3.22356,3.22356,0,0,1,1.19436.491,5.02559,5.02559,0,0,1,.50429.41139,3.38792,3.38792,0,0,1,.96876,2.34892,3.72635,3.72635,0,0,1-.06636.65026,3.37387,3.37387,0,0,1-.18579.62373,4.71469,4.71469,0,0,1-.30522.57064,4.8801,4.8801,0,0,1-.41139.50429,5.02559,5.02559,0,0,1-.50429.41139,3.30547,3.30547,0,0,1-1.84463.55737" transform="translate(-35.5 -118.5)" fill-rule="evenodd"/></svg>
```
Comment (sbdchd) [inline docs/static/img/undraw_docusaurus_tree.svg:1]: I think these undraw images are unused

File: docs/sidebars.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module.exports = {
+  someSidebar: {
+    General: ["quick_start", "safe_migrations", "cli", "github_app"],
+    Rules: [
+      "adding-field-with-default",
```
Comment (sbdchd) [inline docs/sidebars.js:5]: what if we put all the rules related docs in a `//docs/rules/` directory and generated this array by `ls`ing the dir?

File: docs/docs/ban-char-field.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+> increased storage space when using the blank-padded type, and a few extra CPU
+> cycles to check the length when storing into a length-constrained column.
+> While character(n) has performance advantages in some other database systems,
+> there is no such advantage in PostgreSQL; in fact character(n) is usually the
+> slowest of the three because of its additional storage costs. In most
+> situations text or character varying should be used instead.
+
+<https://www.postgresql.org/docs/10/datatype-character.html>
+
+See the `prefer-text-field` rule for info on the advantages of `text` over `varchar`.
```
Comment (sbdchd) [inline docs/docs/ban-char-field.md:20]: link to the other rule?

File: docs/docs/mdx.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+---
+id: mdx
+title: Powered by MDX
+---
+
```
Comment (sbdchd) [inline docs/docs/mdx.md:5]: should this still be here?

File: docs/sidebars.js (hunk 2 above)
Comment (chdsbd) [reply docs/sidebars.js:5]: seems like more work. Maybe something to add later?

File: docs/docs/adding-foreign-key-constraint.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
+---
+id: adding-foreign-key-constraint
+title: adding-foreign-key-constraint
+---
+
+A foreign key constraint should be added with `NOT VALID`.
+
+Adding a foreign key constraint requires a table scan and a SHARE ROW EXCLUSIVE lock on both tables, which blocks writes.
```
Comment (sbdchd) [inline docs/docs/adding-foreign-key-constraint.md:8]:
```suggestion
Adding a foreign key constraint requires a table scan and a `SHARE ROW EXCLUSIVE` lock on both tables, which blocks writes.
```

File: docs/docs/adding-foreign-key-constraint.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+Adding the constraint as NOT VALID in one transaction and then using
```
Comment (sbdchd) [inline docs/docs/adding-foreign-key-constraint.md:10]:
```suggestion
Adding the constraint as `NOT VALID` in one transaction and then using
```

File: docs/docs/adding-foreign-key-constraint.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+VALIDATE in another transaction will allow writes when adding the
```
Comment (sbdchd) [inline docs/docs/adding-foreign-key-constraint.md:11]:
```suggestion
`VALIDATE` in another transaction will allow writes when adding the
```

File: docs/docs/adding-not-nullable-field.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+Use a check constraint instead of setting a column as `NOT NULL`.
+
+## problem
+
+Adding a column as `NOT NULL` requires a table scan and the `ALTER TABLE` requires
+an `ACCESS EXCLUSIVE` lock. Reads and writes will be disabled while this statement is running.
+
+## solution
+
+Add a column as nullable and use a check constraint to verify integrity. The check constraint should be added as NOT NULL and then validated.
```
Comment (sbdchd) [inline docs/docs/adding-not-nullable-field.md:15]:
```suggestion
Add a column as nullable and use a check constraint to verify integrity. The check constraint should be added as `NOT NULL` and then validated.
```

File: docs/docs/adding-not-nullable-field.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+```sql
+ALTER TABLE "core_recipe" ADD COLUMN "foo" integer DEFAULT 10;
+ALTER TABLE "core_recipe" ADD CONSTRAINT foo_not_null
+    CHECK ("foo" IS NOT NULL) NOT VALID;
+-- backfill column so it's not null
+ALTER TABLE "core_recipe" VALIDATE CONSTRAINT foo_not_null;
+```
+
+Add the column as nullable, add a check constraint as NOT VALID that verifies the column is not null, backfill the column so it no longer contains null values, validate the constraint to verify existing rows are valid.
```
Comment (sbdchd) [inline docs/docs/adding-not-nullable-field.md:33]:
```suggestion
Add the column as nullable, add a check constraint as `NOT VALID` that verifies the column is not null, backfill the column so it no longer contains null values, validate the constraint to verify existing rows are valid.
```

File: docs/docs/adding-serial-primary-key-field.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
+---
+id: adding-serial-primary-key-field
+title: adding-serial-primary-key-field
+---
+
+Outlined in [Citus' 2018 post on tips for Postgres
+locking](https://www.citusdata.com/blog/2018/02/22/seven-tips-for-dealing-with-postgres-locks/)
+as well as the Postgres docs, adding a primary key constraint is a blocking
```
Comment (sbdchd) [inline docs/docs/adding-serial-primary-key-field.md:8]:
```suggestion
as well as [the Postgres docs](https://www.postgresql.org/docs/current/sql-altertable.html), adding a primary key constraint is a blocking
```

File: docs/docusaurus.config.js [hunk 11]
```diff
@@ -0,0 +1,95 @@
+module.exports = {
+  title: "Squawk",
+  tagline: "A linter for Postgres migrations",
+  url: "https://your-docusaurus-test-site.com",
```
Comment (sbdchd) [inline docs/docusaurus.config.js:4]: do we need to update this?

**Review by sbdchd — APPROVED** [review]

**Pull request merged**